src/views/RichWorkspace.vue

```diff
@@ -221,6 +190,72 @@ export default {
 			// schedule to normal behaviour
 			this.$_timeoutAutohide = setTimeout(this.onTimeoutAutohide, 7000) // 7s
 		},
+		loadMenuRichWorkspace() {
+			this.getFileInfo()
+				.then((workspaceFileExists) => {
+					const self = this
+					if (!workspaceFileExists) {
+						const newRichWorkspaceFileMenuPlugin = {
+							attach(menu) {
+								const fileList = menu.fileList
+
+								// only attach to main file list, public view is not supported yet
+								if (fileList.id !== 'files' && fileList.id !== 'files.public') {
+									return
+								}
+
+								menu.render = function() {
+									this.$el.html(this.template({
+										uploadMaxHumanFileSize: 'TODO',
+										uploadLabel: t('files', 'Upload file'),
+										items: this._menuItems,
+									}))
+
+									// trigger upload action also with keyboard navigation on enter
+									this.$el.find('[for="file_upload_start"]').on('keyup', function(event) {
+										if (event.key === ' ' || event.key === 'Enter') {
+											document.getElementById('file_upload_start').click()
+										}
+									})
+									this.$el.find('[data-action="rich-workspace-init"]').on('click', function(event) {
+										window.FileList
+											.createFile('Readme.md', { scrollTo: false, animate: false })
+											.then(() => {
+												menu.removeMenuEntry('rich-workspace-init')
+												return self.getFileInfo()
+											})
+										OC.hideMenus()
+									})
+								}
+
+								// register the new menu entry
+								menu.addMenuEntry({
+									id: 'rich-workspace-init',
+									displayName: t('text', 'Add rich workspace'),
+									templateName: t('text', 'Readme') + '.' + loadState('text', 'default_file_extension'),
+									iconClass: 'icon-filetype-text',
+									fileType: 'file',
+									actionHandler() {
+										return window.FileList
+											.createFile('Readme.md', { scrollTo: false, animate: false })
+											.then(() => {
+												menu.removeMenuEntry('rich-workspace-init')
+												return self.getFileInfo()
+											})
+									},
+								})
+							},
+						}
+						OC.Plugins.register('OCA.Files.NewFileMenu', newRichWorkspaceFileMenuPlugin)
```

Comment:
I think this does not work when changing directories as the plugin would be registered each time or the folder info would only be fetched for the first one. 

I think the best would be to move this to https://github.com/nextcloud/text/blob/81d7af739e8c365793888aaa40105ac03f79843e/src/files.js#L60

Reply:
Show dynamically showing or hiding we can probably use the same events as in https://github.com/nextcloud/text/blob/e286194c690c15c5d66b3a7198ee76b9366f1201/src/helpers/files.js#L177-L182

Reply:
I have a question, how can I hide the menu?
- If using Document Select, it won't find the menu because it doesn't show the menu yet if it doesn't click the menu button
- Is there a way to add a Menu item using FileList? I only know how to use OCA.Files.NewFileMenu